### PR TITLE
feat(code-input): add java, go, xml, c#, tsx, ruby, yaml support to code input

### PR DESCRIPTION
**Pull request by skogsmaskin**
This is a follow up on this PR: https://github.com/sanity-io/sanity/pull/1563

Typescript support was already added in the new refactored Code Editor (typescript / sanity-ui).

Added support for java, go, xml, c#, tsx, ruby, yaml.

Also refactored the code so that we don't have to keep two separate language setups for the previews and editor.

Not sure about the increased bundle size from these changes. Maybe we should check that before merging.

### CODE REVIEW

File: packages/@sanity/code-input/src/config.ts [hunk 1]
```diff
@@ -2,20 +2,27 @@ import {CodeInputLanguage} from './types'
 
 export const SUPPORTED_LANGUAGES: CodeInputLanguage[] = [
   {title: 'Batch file', value: 'batchfile'},
+  {title: 'C#', value: 'csharp'},
   {title: 'CSS', value: 'css'},
-  {title: 'SASS', value: 'sass'},
-  {title: 'SCSS', value: 'scss'},
+  {title: 'GoLang', value: 'golang'},
```
Comment (rexxars) [inline packages/@sanity/code-input/src/config.ts:7]:
```suggestion
  {title: 'Go', value: 'golang'},
```
I think Go is usually just referred to as "Go" unless you need to differentiate between the language and the.. game? word?

**Comment by rexxars** [comment]
This PR adds about 126 kB pre-minifying. Thoughts?

**Review by rexxars — CHANGES_REQUESTED** [review]

File: packages/@sanity/code-input/src/config.ts (hunk 1 above)
Comment (skogsmaskin) [reply packages/@sanity/code-input/src/config.ts]: Fixed.

**Comment by skogsmaskin** [comment]
I added Yarn also by popular demand.

**Comment by skogsmaskin** [comment]
> This PR adds about 126 kB pre-minifying. Thoughts?

I think that's ok? I much rather have that than loading it from somewhere async.

File: packages/@sanity/code-input/src/editorSupport.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import 'brace/mode/python'
+import 'brace/mode/ruby'
+import 'brace/mode/sass'
+import 'brace/mode/scss'
+import 'brace/mode/sh'
+import 'brace/mode/text'
+import 'brace/mode/tsx'
+import 'brace/mode/typescript'
+import 'brace/mode/xml'
+import 'brace/mode/yaml'
```
Comment (llibdude) [inline packages/@sanity/code-input/src/editorSupport.ts:25]: I had requested `yaml`  [here](https://github.com/sanity-io/sanity/issues/2682)  so thanks for including it!  I noticed `yaml` isnt referenced in the `config.ts` file above. I don't know much about the codebase but I assume it should be?

File: packages/@sanity/code-input/src/config.ts [hunk 3]
```diff
@@ -1,21 +1,30 @@
...
+  {title: 'Ruby', value: 'ruby'},
+  {title: 'SASS', value: 'sass'},
+  {title: 'SCSS', value: 'scss'},
   {title: 'sh', value: 'sh'},
-  {title: 'Plain text', value: 'text'},
-  {title: 'GROQ', value: 'groq'},
+  {title: 'TSX', value: 'tsx'},
+  {title: 'TypeScript', value: 'typescript'},
+  {title: 'XML', value: 'xml'},
+  {title: 'Yarn', value: 'yarn'},
```
Comment (llibdude) [inline packages/@sanity/code-input/src/config.ts:27]: oh I see, should this be `yaml` not `yarn`?

Comment (skogsmaskin) [reply packages/@sanity/code-input/src/config.ts]: Most def should!

File: packages/@sanity/code-input/src/editorSupport.ts (hunk 2 above)
Comment (skogsmaskin) [reply packages/@sanity/code-input/src/editorSupport.ts:25]: Yep! As you pointed out and now fixed. Thank you.

**Review by rexxars — APPROVED** [review]
Given we don't have time to refactor this with upgraded dependencies and make dynamic loading of languages work, I think this looks good.

**Pull request merged**